src/control/controlframepos.cpp

```diff
@@ -0,0 +1,19 @@
+#include "control/controlframepos.h"
+
+#include "moc_controlframepos.cpp"
+
+ControlFramePos::ControlFramePos(const ConfigKey& key, mixxx::audio::FramePos defaultPosition)
+        : ControlObject(key) {
+    setDefaultPosition(defaultPosition);
+    reset();
+    connect(this,
+            &ControlFramePos::valueChanged,
+            this,
+            &ControlFramePos::slotValueChanged,
+            Qt::DirectConnection);
+}
+
+void ControlFramePos::slotValueChanged(double value) {
+    const auto position = mixxx::audio::FramePos::fromEngineSamplePosMaybeInvalid(value);
+    emit positionChanged(position);
```

Comment:
Even though this is a direct connection is is a connection with a bit of boilerplate that draws CPU. 
I would prefer move the contend of slotValueChanged)() to a virtual function. The base class implementation can than call just emit valueChanged(v); 

This way we have no exrta roundtrip.